athenahealth/appointment_status.go

```diff
@@ -0,0 +1,57 @@
+package athenahealth
+
+import "strconv"
+
+// AppointmentStatus is derived from https://docs.athenahealth.com/api/api-ref/appointment#Book-appointment
+// The athenaNet appointment status. There are several possible statuses.
+// x=cancelled
+// f=future (It can include appointments where were never checked in, even if the appointment date is in the past. It is up to a practice to cancel appointments as a no show when appropriate to do so.)
+// o=open
+// 2=checked in
+// 3=checked out
+// 4=charge entered (i.e. a past appointment)
+type AppointmentStatus string
+
+const (
+	AppointmentStatusCancelled     AppointmentStatus = "x"
+	AppointmentStatusChargeEntered AppointmentStatus = "4"
+	AppointmentStatusCheckedIn     AppointmentStatus = "2"
+	AppointmentStatusCheckedOut    AppointmentStatus = "3"
+	AppointmentStatusFuture        AppointmentStatus = "f"
+	AppointmentStatusOpen          AppointmentStatus = "o"
+)
+
+func (as AppointmentStatus) Int() int {
+	intVal, convErr := strconv.Atoi(string(as))
+	if convErr != nil {
+		return -1
+	}
+
+	return intVal
+}
+
+func (as AppointmentStatus) String() string {
+	if as.Valid() {
+		return string(as)
+	}
+
+	return ""
+}
+
+func (as AppointmentStatus) Valid() bool {
+	appointmentStatuses := []AppointmentStatus{
+		AppointmentStatusCancelled,
+		AppointmentStatusChargeEntered,
+		AppointmentStatusCheckedIn,
+		AppointmentStatusCheckedOut,
+		AppointmentStatusFuture,
+		AppointmentStatusOpen,
+	}
```

Comment:
we generally like to define this list just under the enum/const definitions to improve co-location. If/when a new const is introduced, it's a little clearer to add the new const to the list.

Reply:
I agree however every piece of code is in the same package in this repo so I wasn't sure how else to hide this array which wouldn't be used for anything except for validation. if it's outside the Valid() function it will pollute the `client.appointment...` namespace / autocompletion in people's IDEs. 

I will move it though no sweat.

Reply:
it shouldn't pollute the namespace for clients as it shouldn't be exported

Reply:
I meant for all internal development / users since Appointment... already has a ton of autocomplete options already

Reply:
ah fair - addressed here https://github.com/eleanorhealth/go-athenahealth/pull/105#discussion_r1269697713